pkg/operator/controller/controller_router_deployment.go

```diff
@@ -267,24 +267,14 @@ func deploymentConfigChanged(current, expected *appsv1.Deployment) (bool, *appsv
 		cmp.Equal(current.Spec.Template.Spec.Containers[0].Env, expected.Spec.Template.Spec.Containers[0].Env, cmpopts.EquateEmpty(), cmpopts.SortSlices(cmpEnvs)) &&
 		current.Spec.Template.Spec.Containers[0].Image == expected.Spec.Template.Spec.Containers[0].Image &&
 		current.Spec.Replicas != nil &&
-		*current.Spec.Replicas == *expected.Spec.Replicas {
+		*current.Spec.Replicas == *expected.Spec.Replicas &&
+		current.Spec.Template.Spec.HostNetwork == current.Spec.Template.Spec.HostNetwork {
```

Comment:
We are not comparing `current.Spec.Template.Spec.Containers[0].LivenessProbe` or `current.Spec.Template.Spec.Containers[0].ReadinessProbe`.  However we do not _really_ need to compare `LivenessProbe` or `ReadinessProbe` because they only change when the strategy changes to or from hostnetwork, so if we correctly compare `HostNetwork`, it will act as a proxy for `LivenessProbe` and `ReadinessProbe` (but see the above comment).  Maybe worth a comment?

Reply:
As above re: spec hasher — I _think_ that what we want is a totally generic spec hash comparison to insulate ourselves from this sort of thing. It must be possible, as the upstream deployment controller does it (which is how it can totally generically roll out a new deployment when you change `deployment.spec`). Although that might be based on generation, which means the persistence layer did the hashing... need to dig around and see how this is generally accounted for